pkg/storage/cache_test.go

```diff
@@ -137,20 +137,16 @@ func TestCheckCacheKeyContextualTuplesOrdering(t *testing.T) {
 	require.Equal(t, key1, key2)
 }
 
-func TestCheckCacheKeyContextualTuplesWithConditionsOrdering(t *testing.T) {
+func TestCheckCacheKeyConsidersCondition(t *testing.T) {
```

Comment:
**Request:** Can we also add a test asserting that two requests, one with contextual tuple with and without a condition have unequal cache keys?